lib/arrays.md

```diff
@@ -5,44 +5,83 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is an Array?
 
-Replace me with your response...
-
+    #An Array is an ordered list of objects, you know that it is an array by the 
+    #square brackets.
+    
+    #example: ['dog', 'cat', 'bird', 21]
+    
 # What are some examples of information that would be Arrays as opposed to some other data type?
 
-Replace me with your response...
+    # list of names
+    # list of numbers
+    # scores (if you don't need to relate the score to a name or other object)
+    # prices of a specific object sold by different companies
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+    #You can use element indexes to retrive a given element of an array
+    # example: if x was the given array, to retrive the 6th element of x you would 
+    # do x[5], to retrieve the 8th element of x you would do x[7].
+    #If you try to retrieve an element that doesn't exist from an array then a
+    #blank space would be returned, it would not produce an error.
+    
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+    # Yes it is possible to find the -6th element of an array. Using the same
+    # method as above, x[-6] would return the 6th to last element of a given array.
+    # It is reading from the last element of an array to the first, instead of the
+    # other way around.
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+    # One way to perform an operation on every element inside an Array would be to
+    # use the map method. This will create a new array. It you want it to replace
+    # the elements instead of making a new array use map! method.
+    
+    #example: x = [1,2,3]
+              x.map! { |x| x*2}
```

Comment:
Regarding the differences between `map` and `map!`, why might you want to use one over the other?

When is a time you might use `each` instead of `map` or `map!`?